loadlib/AI/IArtificialIntelligence.hpp

```diff
@@ -0,0 +1,80 @@
+#ifndef __IARTIFICIAL_INTELLIGENCE_HPP__
+#define __IARTIFICIAL_INTELLIGENCE_HPP__
+#include <iostream>
+#include <string>
+#include <math.h>
+
+struct AIPosition
+{
+	int X;
+	int Y;
+};
+
+enum Action
+{
+	DEFAULT,
+	SHOOT,
+	DEAD
+};
+
+struct vec2D
+{
+		float X;
+		float Y;
+};
+
+class IArtificialIntelligence
+{
+public:
+	explicit IArtificialIntelligence() = default;
+	virtual ~IArtificialIntelligence() = default;
+
+	// CORE
+	virtual Action actualize(const uint32_t enX, const uint32_t enY) {return Action::DEFAULT;};
+
+	virtual bool setDamages(uint32_t dmg)
+	{
+			uint32_t l = this->getLife();
+
+			this->setLife(l - dmg);
+			return (this->getLife() > 0);
+	}
+
+	// SETTER | GETTER
+	virtual const AIPosition getPosition() { return _pos; }
+	virtual void setPosition(const AIPosition p) { _pos = p; }
```

Comment:
getter should be const function => virtual const AIPosition getPosition() **_const_** { return _pos; }

Reply:
same for everything below